src/cards/merge.ts

```diff
@@ -0,0 +1,194 @@
+            mergeUnit(target, similarUnits, underworld, prediction);
+          } else if (Pickup.isPickup(target)) {
+            const similarPickups = similarThings as Pickup.IPickup[];
+            mergePickup(target, similarPickups, underworld, prediction);
+            mergedTargets.concat(similarPickups);
+          }
+        }
+      }
+
+      if (!mergedTargets.length) {
+        refundLastSpell(state, prediction, 'Target things of the same type!')
+      }
+      return state;
+    },
+  },
+};
+
+export function mergeUnit(target: Unit.IUnit, unitsToMerge: Unit.IUnit[], underworld: Underworld, prediction: boolean) {
+  // TODO - I forsee a bug that causes modifier addition to change game state before merge is complete
+  // I.E. Target gets a few stacks of suffocate and dies, bloat triggers, bunch of stuff changes.
+  // Loop keeps going and reaches some undefined value, or tries to add modifiers to dead target, etc.
+  for (const unit of unitsToMerge) {
+    // Never merge a player controlled unit
+    if (unit.unitType == UnitType.PLAYER_CONTROLLED) continue;
+
+    // HP / Stam / Mana
+    target.healthMax += unit.healthMax;
+    target.health += unit.health;
+    //target.staminaMax += unit.staminaMax
+    //target.stamina += unit.stamina;
+    target.manaMax += unit.manaMax;
+    target.mana += unit.mana;
+    // Damage / Other
+    target.damage += unit.damage;
+    target.manaCostToCast += unit.manaCostToCast;
+    target.manaPerTurn += unit.manaPerTurn;
+
+    // Modifiers
+    for (const modifierKey of Object.keys(unit.modifiers)) {
+      const modifier = allModifiers[modifierKey];
+      const modifierInstance = unit.modifiers[modifierKey];
+      if (modifier && modifierInstance) {
+        if (modifier?.add) {
+          modifier.add(target, underworld, prediction, modifierInstance.quantity, modifierInstance);
+        }
+      } else {
+        console.error("Modifier doesn't exist? This shouldn't happen.");
+      }
+    }
+
+    // Give XP
+    // Should be done via an OnDeathEvent that merge adds to the target
+    // which calls reportEnemyKilled() once per merged unit
+    if (unit.originalLife) {
+      underworld.enemiesKilled++;
+    }
+
+    // Prediction Lines
+    if (prediction) {
+      const graphics = globalThis.predictionGraphics;
+      if (graphics) {
+        const lineColor = colors.manaBlue;
+        graphics.lineStyle(3, lineColor, 0.7);
+        graphics.moveTo(unit.x, unit.y);
+        graphics.lineTo(target.x, target.y);
+        graphics.drawCircle(target.x, target.y, 3);
+      }
+    }
+
+    Unit.cleanup(unit);
+  }
+}
+
+
+export function mergePickup(target: Pickup.IPickup, pickupsToMerge: Pickup.IPickup[], underworld: Underworld, prediction: boolean) {
+  console.log("TODO - Merge Pickups");
+  for (const pickup of pickupsToMerge) {
+    // TODO - Pickup Merging
+    // Will require pickup rewrite
+
+    // Combine pickup effects / strength?
```

Comment:
Let's make this refactor a part of this spell.
Give pickups a "power" property.  Everything currently starts at "1".  Merged potions get bigger and have their power property summed.  This _should_ be a relatively small refactor I think.

Reply:
Potion effects should be multiplied by their power level just like how 'quantity' works for spells.